### PR TITLE
Adding tests basic getters

### PR DESCRIPTION
**Pull request by Blast545**
First round of tests for increasing coverage of `rclcpp`, those are tackling some basic getter functions, let me know if you think I can easily improve those tests.

In particular, I added this test:

``` cpp
size_t publisher_queue_size = publisher->get_queue_size();
// TODO(blast545): get default rmw qos options to compare here
EXPECT_NE(800u, publisher_queue_size);
```

If you guys know to get default QOS configuration please let me know, otherwise I'll check this out before merging.

### CODE REVIEW

File: rclcpp/test/rclcpp/test_serialized_message.cpp [hunk 1]
```diff
@@ -159,6 +171,82 @@ TEST(TestSerializedMessage, serialization) {
...
+  // Error because it was moved
+  EXPECT_EQ(RCUTILS_RET_INVALID_ARGUMENT, rmw_serialized_message_fini(&rcl_serialized_msg));
+}
+
+TEST(TestSerializedMessage, failed_init_throws) {
+  rclcpp::SerializedMessage serialized_msg(13);
+  EXPECT_EQ(13u, serialized_msg.capacity());
+
+  // Constructor with invalid size throws exception
+  EXPECT_ANY_THROW(rclcpp::SerializedMessage serialized_msg_fail(-1););
```
Comment (brawner) [inline rclcpp/test/rclcpp/test_serialized_message.cpp:235]: `EXPECT_THROW` or `RCLCPP_EXPECT_THROW_EQ`? Same q for line 244

File: rclcpp/test/rclcpp/test_serialized_message.cpp [hunk 2]
```diff
@@ -159,6 +171,82 @@ TEST(TestSerializedMessage, serialization) {
...
+  rclcpp::SerializedMessage serialized_msg_move_rcl(2);
+  EXPECT_EQ(2u, serialized_msg_move_rcl.capacity());
+  EXPECT_EQ(0u, serialized_msg_move_rcl.size());
+  serialized_msg_move_rcl = std::move(rcl_serialized_msg);
+  EXPECT_EQ(13u, serialized_msg_move_rcl.capacity());
+  EXPECT_EQ(content_size, serialized_msg_move_rcl.size());
+  EXPECT_EQ(0u, rcl_serialized_msg.buffer_capacity);
+
+  // Error because it was moved
+  EXPECT_EQ(RCUTILS_RET_INVALID_ARGUMENT, rmw_serialized_message_fini(&rcl_serialized_msg));
```
[the remark this answers is not in the document]
Comment (Blast545) [reply rclcpp/test/rclcpp/test_serialized_message.cpp:227]: I think in this case, the std::move calls [this method](https://github.com/ros2/rclcpp/blob/01d6f52e3281f6df2c191690527f9208006a5c50/rclcpp/src/rclcpp/serialized_message.cpp#L68) defined for the class. So the msg is left in a zero initialized defined state.

File: rclcpp/test/rclcpp/test_publisher.cpp [hunk 3]
```diff
@@ -203,3 +203,57 @@ TEST_F(TestPublisherSub, construction_and_destruction) {
...
+ */
+TEST_F(TestPublisher, basic_getters) {
+  initialize();
+  using test_msgs::msg::Empty;
+  {
+    auto publisher = node->create_publisher<Empty>("topic", 42);
+
+    size_t publisher_queue_size = publisher->get_queue_size();
+    // TODO(blast545): get default rmw qos options to compare here
+    EXPECT_NE(800u, publisher_queue_size);
```
[the remark this answers is not in the document]
Comment (Blast545) [reply rclcpp/test/rclcpp/test_publisher.cpp]: Yes, added 41fcb80

File: rclcpp/test/rclcpp/test_serialized_message.cpp [hunk 4]
```diff
@@ -159,6 +171,82 @@ TEST(TestSerializedMessage, serialization) {
...
+TEST(TestSerializedMessage, assignment_operators) {
+  const std::string content = "Hello World";
+  const auto content_size = content.size() + 1;  // accounting for null terminator
+  auto default_allocator = rcl_get_default_allocator();
+  auto rcl_serialized_msg = rmw_get_zero_initialized_serialized_message();
+  auto ret = rmw_serialized_message_init(&rcl_serialized_msg, 13, &default_allocator);
+  ASSERT_EQ(RCL_RET_OK, ret);
+
+  // manually copy some content
+  std::memcpy(rcl_serialized_msg.buffer, content.c_str(), content.size());
```
[the remark this answers is not in the document]
Comment (Blast545) [reply rclcpp/test/rclcpp/test_serialized_message.cpp]: Yes, changed 15c1b5b

File: rclcpp/test/rclcpp/test_serialized_message.cpp (hunk 1 above)
Comment (Blast545) [reply rclcpp/test/rclcpp/test_serialized_message.cpp]: Yes, changed a85d6c3

**Comment by Blast545** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12247/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7224/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9966/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12140/)

**Comment by brawner** [comment]
Rebuilding macOS:
[Build Status](https://ci.ros2.org/job/ci_osx/9970/)

File: rclcpp/test/rclcpp/test_publisher.cpp [hunk 5]
```diff
@@ -203,3 +203,59 @@ TEST_F(TestPublisherSub, construction_and_destruction) {
     }, rclcpp::exceptions::InvalidTopicNameError);
   }
 }
+
+// Auxiliar class used to test getter for const PublisherBase
```
Comment (brawner) [inline rclcpp/test/rclcpp/test_publisher.cpp:207]:
```suggestion
// Auxiliary class used to test getter for const PublisherBase
```

**Review by brawner — APPROVED** [review]

**Comment by Blast545** [comment]
Green CI + 1 reviewer + PR Job Passing. After CI was checked, b04a1b0 was added. As it only changes a comment, merging.

**Pull request merged**